src/App/Header/UserMenu/UserMenu.tsx

```diff
@@ -0,0 +1,24 @@
+import { Authenticator, useAuthenticator } from "@aws-amplify/ui-react";
+import { Popover } from "@headlessui/react";
+import Icon from "components/Icon";
+import Menu from "./Menu";
+
+export default function UserMenu() {
+  const { user, signOut } = useAuthenticator((context) => [context.route]);
```

Comment:
Bug -  if a user uses the top-right "LOGIN" button for registration, the user appears to be automatically signed in, but when they immediately click on the profile icon in the top right and try to sign out, nothing happens. Furthermore, the profile icon (indicating user is logged in) is present even when navigating to other pages, unless the user navigates by typing the address in the URL bar.

Expectation: when using the top right LOGIN button for registration, the user should not appear to be signed in (when in fact they're not) and they should be immediately asked to confirm their email. This requirement needs to be indicated to the user somehow.

Repro video below (ignore Loom notifications):
https://www.loom.com/share/eb1e1f1752014ea39a16bb96103de8d1?sid=ec35b1b0-db94-4771-8c10-34d63bb5e37f